apps/tenant-management-webapp/src/app/pages/admin/services/comment/comments/addCommentModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FunctionComponent, useState } from 'react';
+
+import { GoAButton, GoAButtonGroup, GoAFormItem, GoAIcon, GoAModal, GoATextArea } from '@abgov/react-components-new';
+import { useDispatch, useSelector } from 'react-redux';
+import { Comment, TopicItem, defaultComment } from '@store/comment/model';
+import { badCharsCheck, wordMaxLengthCheck, isNotEmptyCheck, Validator } from '@lib/validation/checkInput';
+import { useValidators } from '@lib/validation/useValidators';
+import { RootState } from '@store/index';
+import { DescriptionItem, HelpText, ErrorMsg } from '../styled-components';
+interface TopicModalProps {
+  topic: TopicItem;
+  open: boolean;
+  type: string;
+  onCancel?: () => void;
+  onSave: (comment: Comment) => void;
+}
+
+export const AddCommentModal: FunctionComponent<TopicModalProps> = ({
+  topic,
```

Comment:
I think we can have better way to de-couple this modal with the rest of the code. I think we do not need to control the modal open by redux.